twake/backend/node/src/services/workspaces/web/controllers/workspaces.ts

```diff
@@ -124,63 +127,35 @@ export class WorkspacesCrudController
     reply: FastifyReply,
   ): Promise<ResourceGetResponse<WorkspaceObject>> {
     const context = getExecutionContext(request);
-
     const companyUserRole = await this.getCompanyUserRole(context);
 
-    let workspaceEntity: Workspace;
-
-    if (request.params.id) {
-      // ON UPDATE
+    if (!hasCompanyMemberLevel(companyUserRole)) {
+      reply.forbidden(`You are not a member of company ${context.company_id}`);
+      return;
+    }
 
-      workspaceEntity = await this.workspaceService.get({
-        group_id: context.company_id,
-        id: request.params.id,
-      });
+    if (!hasCompanyAdminLevel(companyUserRole) && request.params.id) {
+      const workspaceUserRole = await this.getWorkspaceUserRole(request.params.id, context);
 
-      if (!workspaceEntity) {
-        reply.notFound(`Workspace ${request.params.id} not found`);
+      if (!hasWorkspaceAdminLevel(workspaceUserRole)) {
+        reply.forbidden("You are not a admin of workspace or company");
         return;
       }
+    }
 
-      if (companyUserRole !== "admin") {
-        const workspaceUserRole = await this.getWorkspaceUserRole(request.params.id, context);
-
-        if (workspaceUserRole !== "admin") {
-          reply.forbidden("You are not a admin of workspace or company");
-          return;
-        }
-      }
-
-      const r = request.body.resource;
-
-      workspaceEntity = merge(workspaceEntity, {
-        name: r.name,
-        logo: r.logo,
-        isDefault: r.default,
-        isArchived: r.archived,
-      });
-    } else {
-      // ON CREATE
+    const entity = plainToClass(Workspace, {
+      ...request.body.resource,
+      ...{
+        group_id: request.params.company_id,
+        id: request.params.id,
+      },
+    });
 
-      // you must be a company admin or member to create workspaces
-      if (!["admin", "member"].includes(companyUserRole)) {
-        reply.forbidden(`You are not a member of company ${context.company_id}`);
-        return;
-      }
+    const workspaceEntity = await this.workspaceService
+      .save(entity, {}, context)
+      .then(a => a.entity);
 
-      const obj = merge(new Workspace(), request.body.resource, {
-        group_id: context.company_id,
-        isDefault: request.body.resource.default,
-      });
-
-      workspaceEntity = await this.workspaceService.create(obj).then(a => a.entity);
-      await this.workspaceService.addUser(
-        { id: workspaceEntity.id },
-        { id: context.user.id },
-        "admin",
-      );
-      reply.code(201);
-    }
+    reply.code(201);
```

Comment:
201 is only for created workspace (200 is for updated)

Reply:
done 👍🏻